sqlmesh/core/engine_adapter/base.py

```diff
@@ -118,6 +118,7 @@ def __init__(
         **kwargs: t.Any,
     ):
         self.dialect = dialect.lower() or self.DIALECT
+        self.SCHEMA_DIFFER.dialect = self.dialect
```

Comment:
Is this actually needed?

Reply:
I was thinking initially it was needed since the `SchemaDiffer` is initialised without a dialect, but you're right can remove it

Reply:
I think it's safer to have it? If any downstream adapter doesn't override it

Reply:
I see, fair enough. Do we need to set the dialect in each adapter then? If we choose to mutate `dialect` here, perhaps doing `SchemaDiffer(dialect=DIALECT)` elsewhere is redundant.

Reply:
I was thinking to be on the safe side and cover a case of a SchemaDiffer used by getting it directly of a specific adapter, for example in some tests:
https://github.com/TobikoData/sqlmesh/blob/63e2dbd54430ca16682749ca1c2e5c5f71312303/tests/core/test_plan.py#L1312